File: Strings/palindrome_check.js
function palindromeCheck(string) {
	for (let i = 0; i < Math.floor(string.length / 2); i++) {
		let frontChar = string[i];
		let rearChar = string[string.length - 1 - i];

		if (frontChar !== rearChar) {
			return false;
		}
	}
	return true;
}

// Time Complexity: O(n)
// Space Complexity: O(1)

File: Python/Easy Python/palindrome_check.py
def isPalindrome(string):
    for i in range(int(len(string) / 2)):
        frontChar = string[i]
        rearChar = string[len(string) - 1 - i]

        if frontChar != rearChar:
            return False

    return True

# Time Complexity: O(n)
# Space Complexity: O(1)
